### PR TITLE
Metal binding data, trajectories, and the start of stereochemistry

### PR DESCRIPTION
**Pull request by espottesmith**
This PR _mainly_ adds a new feature/collection/dataset/etc. to MPcules, one describing the coordination behavior and related thermodynamics of metals.

In addition, I have added an API endpoint for molecules/tasks to grab optimization trajectories, similar to how it's done on the materials side, and I've added InChI strings and InChI-keys to molecule and molecule summary docs. These encode some stereo information, which isn't currently captured by the MPculeID.

## Contributor Checklist

- [X] I have run the tests locally and they passed.
- [X] I have added tests, or extended existing tests, to cover any new features or bugs fixed in this PR

### CODE REVIEW

**Comment by espottesmith** [comment]
New Trajectory has been merged in pymatgen, but (I think) no new version has been released. So I guess this needs to wait until that new version happens?

File: emmet-core/emmet/core/qchem/calc_types/utils.py [hunk 1]
```diff
@@ -24,7 +24,7 @@
     "SOLVENT=WATER": "water",
     "SOLVENT=THF": "thf",
```
Comment (orionarcher) [inline emmet-core/emmet/core/qchem/calc_types/utils.py:25]: Can we get rid of these? I'm handling the case of named solvents on the front-end. Let's only include the custom SMD params. e.g.

```python
synonyms = {
    "DIELECTRIC=18,500;N=1,415;ALPHA=0,000;BETA=0,735;GAMMA=20,200;PHI=0,000;PSI=0,143": "3:7 EC:EMC"
    "DIELECTRIC=18,500;N=1,415;ALPHA=0,000;BETA=0,735;GAMMA=20,200;PHI=0,000;PSI=0,000": "3:7 EC:EMC"
}
```

Comment (espottesmith) [reply emmet-core/emmet/core/qchem/calc_types/utils.py]: Done!

**Comment by espottesmith** [comment]
Hey @munrojm (or anyone else): any idea why test_electronic_structure.py is failing? Seems there's an issue deep up the stack, at the pydash level?

**Comment by munrojm** [comment]
Hm, not too sure. I will take a look.

**Comment by espottesmith** [comment]
Hey, any update on this?

**Comment by munrojm** [comment]
Is it still an issue? `FAILED emmet-builders/tests/test_qchem.py::test_molecules_builder` appears to be the failing test at the moment. The ES builder tests seem okay.

**Comment by espottesmith** [comment]
No, seems to have resolved itself.

**Pull request merged**